### ISSUE TITLE
httpd 2.2: Install more modules

### ISSUE THREAD

**Issue by carletes** [docker-library/httpd#6]
Hi! I would like to have the following modules in the `httpd:2.2` image:

* mod_headers.so
* mod_proxy.so
* mod_proxy_http.so
* mod_rewrite.so
* mod_ssl.so

I could do a pull request to modify `Dockerfile` accordingly, so that they get compiled in the `httpd` binary -- or, perhaps better yet, compile *all* Apache modules as shared objects, and let users load them in `httpd.conf` if they are needed.

Any thoughts on this?

**Comment by yosifkit** [comment]
I was going to say yes, but it looks like the default is to [compile `most`](https://httpd.apache.org/docs/2.4/programs/configure.html#otheroptfeat) of them as shared objects.  So I think all of those that you want are there and just need to be enabled in httpd.conf (`usr/local/apache2/conf/httpd.conf`).

```console
root@6fe22a08f435:/usr/local/apache2# ls modules/
httpd.exp		mod_dav_fs.so		    mod_proxy_fcgi.so
mod_access_compat.so	mod_dbd.so		    mod_proxy_ftp.so
mod_actions.so		mod_deflate.so		    mod_proxy_http.so
mod_alias.so		mod_dir.so		    mod_proxy_scgi.so
mod_allowmethods.so	mod_dumpio.so		    mod_proxy_wstunnel.so
mod_auth_basic.so	mod_env.so		    mod_ratelimit.so
mod_auth_digest.so	mod_expires.so		    mod_remoteip.so
mod_auth_form.so	mod_ext_filter.so	    mod_reqtimeout.so
mod_authn_anon.so	mod_file_cache.so	    mod_request.so
mod_authn_core.so	mod_filter.so		    mod_rewrite.so
mod_authn_dbd.so	mod_headers.so		    mod_sed.so
mod_authn_dbm.so	mod_include.so		    mod_session.so
mod_authn_file.so	mod_info.so		    mod_session_cookie.so
mod_authn_socache.so	mod_lbmethod_bybusyness.so  mod_session_crypto.so
mod_authnz_ldap.so	mod_lbmethod_byrequests.so  mod_session_dbd.so
mod_authz_core.so	mod_lbmethod_bytraffic.so   mod_setenvif.so
mod_authz_dbd.so	mod_lbmethod_heartbeat.so   mod_slotmem_shm.so
mod_authz_dbm.so	mod_ldap.so		    mod_socache_dbm.so
mod_authz_groupfile.so	mod_log_config.so	    mod_socache_memcache.so
mod_authz_host.so	mod_log_debug.so	    mod_socache_shmcb.so
mod_authz_owner.so	mod_logio.so		    mod_speling.so
mod_authz_user.so	mod_macro.so		    mod_ssl.so
mod_autoindex.so	mod_mime.so		    mod_status.so
mod_buffer.so		mod_negotiation.so	    mod_substitute.so
mod_cache.so		mod_proxy.so		    mod_unique_id.so
mod_cache_disk.so	mod_proxy_ajp.so	    mod_unixd.so
mod_cache_socache.so	mod_proxy_balancer.so	    mod_userdir.so
mod_cgid.so		mod_proxy_connect.so	    mod_version.so
mod_dav.so		mod_proxy_express.so	    mod_vhost_alias.so
```

These are the ones enabled by default:
```console
root@6fe22a08f435:/usr/local/apache2# httpd -t -D DUMP_MODULES
AH00558: httpd: Could not reliably determine the server's fully qualified domain name, using 172.17.0.143. Set the 'ServerName' directive globally to suppress this message
Loaded Modules:
 core_module (static)
 so_module (static)
 http_module (static)
 mpm_event_module (static)
 authn_file_module (shared)
 authn_core_module (shared)
 authz_host_module (shared)
 authz_groupfile_module (shared)
 authz_user_module (shared)
 authz_core_module (shared)
 access_compat_module (shared)
 auth_basic_module (shared)
 reqtimeout_module (shared)
 filter_module (shared)
 mime_module (shared)
 log_config_module (shared)
 env_module (shared)
 headers_module (shared)
 setenvif_module (shared)
 version_module (shared)
 unixd_module (shared)
 status_module (shared)
 autoindex_module (shared)
 dir_module (shared)
 alias_module (shared)
```

**Comment by carletes** [comment]
Thanks, @yosifkit! I must be doing something wrong, because I don't see any modules on *2.2*:

```
$ docker run -it --rm httpd:2.2 ls -l modules/
Unable to find image 'httpd:2.2' locally
2.2: Pulling from library/httpd
809f35cec54c: Pull complete 
ffc439de6236: Pull complete 
09724b9a21ac: Pull complete 
2ef4a4987dde: Pull complete 
036cd26a362d: Pull complete 
6a625f467b06: Pull complete 
9390ddce8d3e: Pull complete 
1726705c908d: Pull complete 
ff7a79fd516f: Pull complete 
0a1f669f88da: Pull complete 
54b739d5d259: Pull complete 
5935cfcd1d7c: Pull complete 
e2a4fb18da48: Already exists 
58016a5acc80: Already exists 
Digest: sha256:98f072a27fcf4231ed472b1cee5819b7e578b12620deb0611bed2a73a349e9d3
Status: Downloaded newer image for httpd:2.2
total 12
-rw-r--r-- 1 root root 9280 Oct 13 23:49 httpd.exp
$ docker images httpd
REPOSITORY          TAG                 IMAGE ID            CREATED             VIRTUAL SIZE
httpd               2.2                 5935cfcd1d7c        6 days ago          152.9 MB
$
```

On *2.4*, however, things are as you say:

```
(webdev)carlos@joselito-vm:~/src/webdev$ docker run -it --rm httpd:2.4 ls -l modules/
Unable to find image 'httpd:2.4' locally
2.4: Pulling from library/httpd
8872c72273b8: Pull complete 
94876270016d: Pull complete 
d9b06d6ab17d: Pull complete 
c15733c21dc6: Pull complete 
7d2666b99dfb: Pull complete 
f42a11bb13de: Pull complete 
9e77a4c0e6e1: Pull complete 
e2a4fb18da48: Already exists 
58016a5acc80: Already exists 
809f35cec54c: Already exists 
ffc439de6236: Already exists 
09724b9a21ac: Already exists 
2ef4a4987dde: Already exists 
036cd26a362d: Already exists 
Digest: sha256:e03d663d8637abb292e63242e8fa19e38341591fcb0b7b0adf166a9e5fb20278
Status: Downloaded newer image for httpd:2.4
total 3140
-rw-r--r-- 1 root root      14699 Oct 14 20:03 httpd.exp
-rwxr-xr-x 1 root www-data  14504 Oct 14 20:03 mod_access_compat.so
-rwxr-xr-x 1 root www-data  11752 Oct 14 20:04 mod_actions.so
-rwxr-xr-x 1 root www-data  19480 Oct 14 20:04 mod_alias.so
-rwxr-xr-x 1 root www-data   9592 Oct 14 20:03 mod_allowmethods.so
-rwxr-xr-x 1 root www-data  19184 Oct 14 20:03 mod_auth_basic.so
[..]
```

**Comment by 4devnull** [comment]
Hello, 
Do you have find solution ?
Thanx

**Comment by ain** [comment]
For me the latest image is similarly empty and I think it pretty much renders the `2.2` image useless:

```
root@67c46ff279f9:/usr/local/apache2/modules# ls -la 
total 19280
drwxr-sr-x  2 root     www-data     4096 Dec  4 22:30 .
drwxr-sr-x 21 www-data www-data     4096 Dec  5 19:23 ..
-rw-r--r--  1 root     root         9280 Nov 20 06:28 httpd.exp
```

**Comment by ain** [comment]
`mod_rewrite` is a must and it's not there for `2.2`. I'd suggest to definitely add it. `modules` folder as it stands is currently empty.

**Comment by yosifkit** [comment]
I made a PR to fix `2.2` having no modules: #8.  Let me know if the new list of modules looks right.
